e2e/awslogs_test.go

```diff
@@ -63,45 +50,26 @@ var testAwslogs = func() {
 				awslogsStreamKey:              testAwsLogsStream,
 			}
 			creator := cio.BinaryIO(*Binary, args)
-			// Create a new client connected to the containerd daemon
-			client, err := containerd.New(containerdAddress)
-			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-			defer client.Close()
-			// Create a new context with a customized namespace
-			ctx := namespaces.WithNamespace(context.Background(), "testAwslogs")
-			// Pull an image
-			image, err := client.Pull(ctx, testImage, containerd.WithPullUnpack)
-			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-			// Create a new container with the pulled image
-			container, err := client.NewContainer(ctx, testContainerId, containerd.WithImage(image),
-				containerd.WithNewSnapshot("test-snapshot", image), containerd.WithNewSpec(oci.WithImageConfig(image),
-					oci.WithProcessArgs("/bin/sh", "-c", fmt.Sprintf("echo '%s'", testAwsLogsMessage))))
-			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-			defer container.Delete(ctx, containerd.WithSnapshotCleanup) //nolint:errcheck // testing only
-			// Create a new task from the container and start it
-			task, err := container.NewTask(ctx, creator)
-			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-			defer task.Delete(ctx) //nolint:errcheck // testing only
-
-			err = task.Start(ctx)
-			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			sendTestLogByContainerd(creator, testLog)
+			validateTestLogsInAwslogs(cwClient, testAwsLogsGroup, testAwsLogsStream, testLog)
+		})
+	})
+}
 
-			statusC, err := task.Wait(ctx)
-			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-			// Waiting for the task to finish
-			status := <-statusC
-			code, _, err := status.Result()
-			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-			gomega.Expect(code).Should(gomega.Equal(uint32(0)))
+func validateTestLogsInAwslogs(client *cloudwatchlogs.Client, logGroupName string, logStreamName string, testLog string) {
+	cwOutput, err := client.GetLogEvents(context.TODO(), &cloudwatchlogs.GetLogEventsInput{
```

Comment:
Consider passing in a Ginkgo `SpecContext` so that this function can time out when the test times out. Same for `cleanupAwslogs`.

Probably not 100% necessary.

Reply:
Looking into the documentation. https://onsi.github.io/ginkgo/#spec-subjects-it

SpecContext is added immediately after ginkgo.It() such as 
```
ginkgo.It("should send logs to fluentd log driver", func(ctx SpecContext) {}
```

It is usually used for long-running code so may not need for our case.